torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -655,6 +655,61 @@ at::Tensor& AtenXlaType::bernoulli_(at::Tensor& self, const at::Tensor& p,
   return self;
 }
 
+at::Tensor AtenXlaType::binary_cross_entropy(const at::Tensor& self,
+                                             const at::Tensor& target,
+                                             const at::Tensor& weight,
+                                             int64_t reduction) {
+  XLA_FN_COUNTER("xla::");
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
+  XLATensor weight_tensor =
+      bridge::GetOrCreateXlaTensor(weight, self_tensor.GetDevice());
+  return bridge::AtenFromXlaTensor(XLATensor::binary_cross_entropy(
+      self_tensor, bridge::GetXlaTensor(target), weight_tensor, reduction));
+}
+
+at::Tensor AtenXlaType::binary_cross_entropy_backward(
+    const at::Tensor& grad_output, const at::Tensor& self,
+    const at::Tensor& target, const at::Tensor& weight, int64_t reduction) {
+  XLA_FN_COUNTER("xla::");
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
+  XLATensor weight_tensor =
+      bridge::GetOrCreateXlaTensor(weight, self_tensor.GetDevice());
+  return bridge::AtenFromXlaTensor(XLATensor::binary_cross_entropy_backward(
+      bridge::GetXlaTensor(grad_output), self_tensor,
+      bridge::GetXlaTensor(target), weight_tensor, reduction));
+}
+
+at::Tensor& AtenXlaType::binary_cross_entropy_backward_out(
```

Comment:
We cannot move forward and have two implementations of every operator 😐 
Who is calling the out function in these cases?
Otherwise we need to put in place something which allow us to implement only the _out one, and route the other one to the _out one.

Reply:
@dlibenzi In this case it's special because both `bce` and its `out` version are still in our legacy THNN code. So both of them are implemented through one implementation. When they're moved into aten in the future, I think the `out` version will be kept as leaf. But currently when both `out` and normal version can be overrided, overriding normal ones are sufficient.

Reply:
Wait ... can't we autogenerate them with `_FN_OUT` in `gen.py`?
I forgot about that 😁

Reply:
Oh! Nice catch!